File: library/src/main/java/com/smartnsoft/smartappupdate/UpdateScreenAnalyticsInterface.java
package com.smartnsoft.smartappupdate;

import android.os.Bundle;

/**
 * @author Adrien Vitti
 * @since 2018.03.05
 */

public interface UpdateScreenAnalyticsInterface
{

  // region Screen display

  void sendUpdateInfoDisplayed();

  void sendRecomendedUpdateDisplayed();

  void sendBlockingUpdateDisplayed();

  // endregion Screen display

  // region Action Events

  void sendAskLaterEvent();

  void closeAppWithoutUpdateEvent();

  void sendInformativeActionButtonEvent();

  void sendRecommendedActionButtonEvent();

  void sendBlockingActionButtonEvent();

  // endregion Action Events

  int getVersionCode();

  String getDateForAnalytics();

  Bundle generateAnalyticsExtraInfos();

}

File: library/src/main/java/com/smartnsoft/smartappupdate/UpdateScreenAnalyticsInterface.kt
package com.smartnsoft.smartappupdate

import android.os.Bundle

/**
 * @author Adrien Vitti
 * @since 2018.03.05
 */

interface UpdateScreenAnalyticsInterface
{

  // region Screen display

  fun sendUpdateInfoDisplayed()

  fun sendRecomendedUpdateDisplayed()

  fun sendBlockingUpdateDisplayed()

  // endregion Screen display

  // region Action Events

  fun sendAskLaterEvent()

  fun closeAppWithoutUpdateEvent()

  fun sendInformativeActionButtonEvent()

  fun sendRecommendedActionButtonEvent()

  fun sendBlockingActionButtonEvent()

  // endregion Action Events

  fun getVersionCode(): Int

  fun getDateForAnalytics(): String?

  fun generateAnalyticsExtraInfos(): Bundle?

}
